### PR TITLE
issue771 fanCoil unit

### PR DESCRIPTION
**Pull request by icupeiro**
This solves #771 
I will open the pull request already and comment on the work done so far

Current work:

-Created a FanCoilUnits package in IDEAS.Fluid.HeatExchangers.FanCoilUnits
-Created models for 2pipe fan coils: the models are controlled by the air fan, and they DO NOT include a 3-way valve at the exit of the coil.
-BaseClasses with a PartialFanCoil (includes air boundary, air fan and psychometrics in case of a 2-PipeCooling or 4-Pipe type) and a Cooling Coil: The cooling coil uses two NTU models from the Buildings library, one that computes the sensible heat load (transferred to the air stream) and one that computes the total heat load (transferred to the water stream). The model DOES NOT account for the change in condensation in the zone (we assume it is negligible) to avoid algebraic loops by connecting multiple components to the zone air ports. Instead, a heat port is connected to the zone model. IMPORTANT NOTE: A slight modiffication has been done in the model Buildings.Fluid.HeatExchangers.BaseClasses.PartialEffectiveness to change the capacity flow in function of whether there is condensation or not. The variable C1_flow is commented in the equations section
-Type package: TwoPipeHea, TwoPipeCoo, FourPipe
-Validation tests according to manufacturer data from Jaga - model Briza: Results show errors of less than 10%, except for the case where there is condensation and the air flow is much less than then nominal one. The error can scale up to 15-20% here.
-Example with zone, simple hysteresis controller

Further work:

-Clean up the models
-4pipe model
-Include electricity consumption of the fans
-Example with 4pipe model
-mos files for UnitTests
-Documentation
-Revision history

Future implementation:

-I would say to do the implementation WITHIN the zone model in a separate pull request

### CODE REVIEW

**Comment by Mathadon** [comment]
@icupeiro I did some quick checks of the code without looking at details yet. Can you already:
 - [ ] Add a comment that describes the component for each component, medium or variable declaration.
- [ ] Declare parameters at the top of each model. Don't use the `SI` import.
- [ ] Remove `import IDEAS` and `import Buildings`.
- [ ] Copy `Buildings.Fluid.HeatExchangers.DryCoilEffectivenessNTU` to IDEAS.
- [ ] Use IDEAS versions of the models from the `IBPSA` library that you use.

Thanks!

**Comment by icupeiro** [comment]
First round of changes done

**Comment by Mathadon** [comment]
@icupeiro you need my review on this, right?

**Comment by icupeiro** [comment]
@Mathadon yup

**Comment by Mathadon** [comment]
@icupeiro can you fix the errors that are reported in https://travis-ci.org/open-ideas/IDEAS/jobs/496064701 ?
regarding the tolerance, you have to save a tolerance in the simulation model and the script nowadays.. E.g. set the tolerance to 1e-6 in dymola and store that option.

**Comment by Mathadon** [comment]
Some more issues:
 - [ ] `IDEAS.Fluid.HeatExchangers.DryCoilEffectivenessNTU` still uses multiple models of `Buildings`. Those have to be copied, including unit tests and reference results.

**Comment by icupeiro** [comment]
checked

**Comment by Mathadon** [comment]
@icupeiro can you add `WetCoilEffectivenessNTU` and make sure that unit tests run? Then I'm sure that all required models are included in IDEAS and I don't have to check it manually.

**Comment by Mathadon** [comment]
@icupeiro nvm merging all unit tests. I'll do it when we merge the FCU model into IBPSA.

**Comment by icupeiro** [comment]
@Mathadon implementation of the "NTUWet" model done. To speed up the process I will implement the 4pipe model

File: IDEAS/Fluid/HeatExchangers/WetCoilEffectivenessNTU.mo [hunk 1]
```diff
@@ -36,48 +36,13 @@ equation
...
-For a plate exchanger, use
-<a href=\"modelica://Buildings.Fluid.MassExchangers.PlateHeatExchangerEffectivenessNTU\">
-Buildings.Fluid.MassExchangers.PlateHeatExchangerEffectivenessNTU</a>.
-</p>
-<p>
-For a heat and moisture exchanger, use
-<a href=\"modelica://Buildings.Fluid.MassExchangers.ConstantEffectiveness\">
-Buildings.Fluid.MassExchangers.ConstantEffectiveness</a>.
-</p>
+<p>Model of a coil with humidity condensation. This model transfers heat in the amount of </p>
```
Comment (Mathadon) [inline IDEAS/Fluid/HeatExchangers/WetCoilEffectivenessNTU.mo]: @icupeiro can you reformat the documentation again as it used to be?

**Comment by icupeiro** [comment]
@Mathadon 
4pipe model implemented, updated unit tests + rev. history + doc.

**Comment by Mathadon** [comment]
@icupeiro the unit tests still cause errors, can you check them already?

**Comment by Mathadon** [comment]
@icupeiro Thanks for the changes! I went through some of the code again and tried to test it. There are still some more things:
 - [ ] `IDEAS.Fluid.HeatExchangers.FanCoilUnits.TwoPipeCoo` does not work -> hence the question for adding unit tests. Please add an example for every model. 
 - [ ] In this example, please test the actual model. Right now you use the model together with a zone in an on/off fashion, which means that the FanCoil model itself is actually not tested that well. Instead, please test the model standalone, without a zone and test it for a changing inlet temperature and/or mass flow rate. You should then see a smooth transition between the regime with and without condensation and this should be demonstrated in the test. E.g. see attached.
schermafbeelding 2019-02-27 om 08 01 41
 - [ ] Can you include a reference to the source that you used for implementing the model?
 - [ ] IBPSA convention is not to mix graphical components and equations. Therefore, can you modify `IDEAS.Fluid.HeatExchangers.BaseClasses.PartialEffectivenessNTUWet` such that it does not contain graphical components and directly use the functions that those blocks implement instead?
 - [ ] For models that use a zone, don't forget to include a `SimInfoManager`.
 - [ ] Would be good to implement an example that compares the `IDEAS.Fluid.HeatExchangers.DryCoilEffectivenessNTU` with `IDEAS.Fluid.HeatExchangers.WetCoilEffectivenessNTU` in the dry operation mode that verifies that the results are identical for the whole operation regime.

**Comment by icupeiro** [comment]
changes made, waiting for unitTests
for the mixing of graphical components and equations, I substituted most of them with functions with the exception of the computation of the wetBulb temperature, which was indeed a bit more tricky than a standard function. I removed the graphical part though

**Comment by damienpicard** [comment]
@Mathadon and @icupeiro : we need this model by the beginning of July. Can we finish it?

**Comment by icupeiro** [comment]
@damienpicard we are pending some reviews in IBPSA to the wet coil model that can take a while, considering that the model needs to be extended to both sides
In the meanwhile @Mathadon distributed an early version of the model among the partners
I would suggest to extend the hG AHU models library that I created two days ago and include this working version. Perhaps I can do it by the end of the day

**Comment by Mathadon** [comment]
Development continued within IBPSA Project 1.

**Pull request closed without merging**